File: ModuleBase/src/main/java/cn/mycommons/modulebase/IModuleConfig.java
package cn.mycommons.modulebase;

import android.app.Activity;

/**
 * IModuleConfig <br/>
 * Created by xiaqiulei on 2017-05-14.
 */
public interface IModuleConfig {

    void registerRouter(String uri, Class<? extends Activity> activityClass);

    Class<? extends Activity> getRouterActivity(String uri);

    void registerRouter(String uri, IRouterProcess routerProcess);

    IRouterProcess getRouterProcess(String uri);

    <T> void registerService(Class<T> serviceClass, Class<? extends T> implementClass);

    <T> Class<? extends T> getServiceImplementClass(Class<T> serviceClass);
}

File: TestModule/ModuleBase/src/main/java/cn/mycommons/modulebase/IModuleConfig.kt
package cn.mycommons.modulebase

import android.app.Activity

/**
 * IModuleConfig <br></br>
 * Created by xiaqiulei on 2017-05-14.
 */
interface IModuleConfig {

    fun registerRouter(uri: String, activityClass: Class<out Activity>)

    fun getRouterActivity(uri: String): Class<out Activity>?

    fun registerRouter(uri: String, routerProcess: IRouterProcess)

    fun getRouterProcess(uri: String): IRouterProcess?

    fun <T> registerService(serviceClass: Class<T>, implementClass: Class<out T>)

    fun <T> getServiceImplementClass(serviceClass: Class<T>): Class<out T>?
}